src/oidcendpoint/oidc/token_coop.py

```diff
@@ -165,6 +172,14 @@ def _access_token_post_parse_request(self, request, client_id="", **kwargs):
 
         return request
 
+    def check_grant_type_allowed(self, grant_type):
+        grant_types_supported = self.endpoint_context.provider_info['grant_types_supported']
+        if (
+            grant_type not in self.token_grant_types
+            or grant_type not in grant_types_supported
+        ):
+            raise ProcessError("Invalid grant type")
```

Comment:
Add the `grant_type` value to the exception?

Reply:
yes